app/uk/gov/hmrc/gform/validation/ComponentValidator.scala

```diff
@@ -654,6 +673,165 @@ object ComponentValidator {
+        )
+      case (_, Some(errorShortNameStart)) =>
+        validationFailure(
+          fieldValue,
+          genericSterlingErrorPatternStart,
+          Some(List(errorShortNameStart.value(), errorExample.value()))
+        )
+    }
+
+  private def maxDigitSterlingFailure(
+    fieldValue: FormComponent,
+    value: String,
+    maxWhole: Int,
+    errorExample: SmartString
+  )(implicit
+    messages: Messages,
+    sse: SmartStringEvaluator
+  ): ValidatedType[Unit] =
+    fieldValue.errorShortNameStart match {
+      case None =>
+        validationFailure(
+          fieldValue,
+          genericSterlingErrorMaxdigitPattern,
+          Some(
+            List(
+              SmartString.blank.trasform(_ => "Amount", _ => "swm").value(),
+              errorExample.value(),
+              maxWhole.toString
+            )
+          )
+        )
+      case Some(errorShortNameStart) =>
+        validationFailure(
+          fieldValue,
+          genericSterlingErrorMaxdigitPattern,
+          Some(
+            List(
+              errorShortNameStart.value(),
+              errorExample.value(),
+              maxWhole.toString
+            )
+          )
+        )
+    }
+
+  private def wholeSterlingFailure(
+    fieldValue: FormComponent,
+    value: String,
+    errorExample: SmartString
+  )(implicit
+    messages: Messages,
+    sse: SmartStringEvaluator
+  ): ValidatedType[Unit] =
+    fieldValue.errorShortNameStart match {
+      case None =>
+        validationFailure(
+          fieldValue,
+          genericWholesterlingErrorPencePattern,
+          Some(
+            List(
+              SmartString.blank.trasform(_ => "Amount", _ => "swm").value(),
+              errorExample.value()
+            )
+          )
+        )
+      case Some(errorShortNameStart) =>
+        validationFailure(
+          fieldValue,
+          genericWholesterlingErrorPencePattern,
+          Some(
+            List(
+              errorShortNameStart.value(),
+              errorExample.value()
+            )
+          )
+        )
+    }
+
+  private def positiveSterlingFailure(
+    fieldValue: FormComponent,
+    value: String,
+    errorExample: SmartString
+  )(implicit
+    messages: Messages,
+    sse: SmartStringEvaluator
+  ): ValidatedType[Unit] =
+    fieldValue.errorShortNameStart match {
+      case None =>
+        validationFailure(
+          fieldValue,
+          genericPositiveSterlingErrorPositivePattern,
+          Some(
+            List(
+              SmartString.blank.trasform(_ => "Amount", _ => "swm").value(),
+              errorExample.value()
+            )
+          )
+        )
+      case Some(errorShortNameStart) =>
+        validationFailure(
+          fieldValue,
+          genericPositiveSterlingErrorPositivePattern,
+          Some(
+            List(
+              errorShortNameStart.value(),
+              errorExample.value()
+            )
+          )
+        )
+    }
```

Comment:
```suggestion
validationFailure(
      fieldValue,
      genericPositiveSterlingErrorPositivePattern,
      Some(
        List(
          fieldValue.errorShortNameStart match {
            case None                      => SmartString.blank.trasform(_ => "Amount", _ => "swm").value()
            case Some(errorShortNameStart) => errorShortNameStart.value()
          },
          errorExample.value()
        )
      )
    )
```

Why not move pattern matching where it really make a difference? 

Please inspect other new methods if similar improvement is possible.